shared/p2p/service.go

```diff
@@ -173,29 +178,37 @@ func (s *Server) Start() {
 	defer span.End()
 	log.Info("Starting service")
 
-	if !s.noDiscovery && s.bootstrapNode != "" {
-		if err := startDHTDiscovery(ctx, s.host, s.bootstrapNode); err != nil {
-			log.Errorf("Could not start peer discovery via DHT: %v", err)
+	if !s.noDiscovery {
+		if s.bootstrapNode != "" {
+			if err := startDHTDiscovery(ctx, s.host, s.bootstrapNode); err != nil {
+				log.Errorf("Could not start peer discovery via DHT: %v", err)
+			}
+			bcfg := kaddht.DefaultBootstrapConfig
+			bcfg.Period = time.Duration(30 * time.Second)
+			if err := s.dht.BootstrapWithConfig(ctx, bcfg); err != nil {
+				log.Errorf("Failed to bootstrap DHT: %v", err)
+			}
 		}
-		bcfg := kaddht.DefaultBootstrapConfig
-		bcfg.Period = time.Duration(30 * time.Second)
-		if err := s.dht.BootstrapWithConfig(ctx, bcfg); err != nil {
-			log.Errorf("Failed to bootstrap DHT: %v", err)
+		if s.relayNodeAddr != "" {
+			if err := dialRelayNode(ctx, s.host, s.relayNodeAddr); err != nil {
+				log.Errorf("Could not dial relay node: %v", err)
+			}
 		}
-	}
-	if !s.noDiscovery && s.relayNodeAddr != "" {
-		if err := dialRelayNode(ctx, s.host, s.relayNodeAddr); err != nil {
-			log.Errorf("Could not dial relay node: %v", err)
+
+		if err := startmDNSDiscovery(ctx, s.host); err != nil {
+			log.Errorf("Could not start peer discovery via mDNS: %v", err)
 		}
-	}
 
-	if err := startmDNSDiscovery(ctx, s.host); err != nil {
-		log.Errorf("Could not start peer discovery via mDNS: %v", err)
-		return
+		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
 	}
 
-	if !s.noDiscovery {
-		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
+	for _, peer := range s.staticPeers {
+		peerInfo, err := peerInfoFromAddr(peer)
+		if err != nil {
+			log.Errorf("Invalid peer address: %v", err)
+		} else {
+			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, oneHour)
```

Comment:
Why one hour? Shouldn't it be indefinitely?

Reply:
I profess my ignorance of the API here. I assumed a TTL was mandatory.

Reply:
It does seem like a TTL is mandatory.

Reply:
```suggestion
			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, 24 * time.Hour)
```
Yeah, unfortunate design of the API that you can't set a non-expiring TTL. If you provide `0` then it won't add the address at all. Let's set 24hr and assume that is enough